In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('transaction_data.csv')

In [3]:
# reading first 10 columns
df.head(10)

TransactionID  CustID       Date       Employee  ProductID  \
0          10258       1  7/17/2023  Nancy Davolio         32   
1          10275       2   8/7/2023  Nancy Davolio         24   
2          10275       2   8/7/2023  Nancy Davolio         59   
3          10292       3  8/28/2023  Nancy Davolio         20   
4          10293       4  8/29/2023  Nancy Davolio         18   
5          10293       4  8/29/2023  Nancy Davolio         24   
6          10293       4  8/29/2023  Nancy Davolio         63   
7          10293       4  8/29/2023  Nancy Davolio         75   
8          10304       4  9/12/2023  Nancy Davolio         59   
9          10304       4  9/12/2023  Nancy Davolio         71   

              ProductName  VendorID                    ProductVendor  \
0      Mascarpone Fabioli        14          Formaggi Fortini s.r.l.   
1        Guaran Fantstica        10        Refrescos Americanas LTDA   
2    Raclette Courdavault        28                      Gai pturage   
3  Sir Rodney's Marmalade         8         Specialty Biscuits, Ltd.   
4        Carnarvon Tigers         7                    Pavlova, Ltd.   
5        Guaran Fantstica        10        Refrescos Americanas LTDA   
6            Vegie-spread         7                    Pavlova, Ltd.   
7      Rhnbru Klosterbier        12  Plutzer Lebensmittelgromrkte AG   
8    Raclette Courdavault        28                      Gai pturage   
9             Flotemysost        15                  Norske Meierier   

   UnitPrice  Quantity  Subtotal  Tax(8%)  OrderTotal  
0       25.6         6     153.6    12.29      165.89  
1        3.6        12      43.2     3.46       46.66  
2       44.0         6     264.0    21.12      285.12  
3       64.8        20    1296.0   103.68     1399.68  
4       50.0        12     600.0    48.00      648.00  
5        3.6        10      36.0     2.88       38.88  
6       35.1         5     175.5    14.04      189.54  
7        6.2         6      37.2     2.98       40.18  
8       44.0        10     440.0    35.20      475.20  
9       17.2         2      34.4     2.75       37.15

In [4]:
df.describe()

TransactionID       CustID    ProductID     VendorID    UnitPrice  \
count    1353.000000  1353.000000  1353.000000  1353.000000  1353.000000   
mean    10661.179601    38.942350    40.389505    14.318551    25.908899   
std       243.094311    25.056615    22.277628     8.217175    28.613063   
min     10248.000000     1.000000     1.000000     1.000000     2.000000   
25%     10454.000000    17.000000    21.000000     7.000000    12.000000   
50%     10656.000000    35.000000    41.000000    14.000000    18.400000   
75%     10864.000000    60.000000    60.000000    21.000000    31.230000   
max     11077.000000    89.000000    77.000000    29.000000   263.500000   

          Quantity     Subtotal      Tax(8%)   OrderTotal  
count  1353.000000  1353.000000  1353.000000  1353.000000  
mean     12.631929   324.537568    25.962971   350.500540  
std       6.395421   403.982507    32.318600   436.301107  
min       1.000000     4.800000     0.380000     5.180000  
25%       7.000000   100.000000     8.000000   108.000000  
50%      12.000000   210.000000    16.800000   226.800000  
75%      20.000000   394.000000    31.520000   425.520000  
max      24.000000  4216.000000   337.280000  4553.280000

***
This shows great insight for unitprice, qunatity, subtotal, and order total for mean, max, min and everything in between

Something that stands out to me is the subtotal column for transactions:

Avg: 324.54

Max: 4216
***

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1353 entries, 0 to 1352
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   TransactionID  1353 non-null   int64  
 1   CustID         1353 non-null   int64  
 2   Date           1353 non-null   object 
 3   Employee       1353 non-null   object 
 4   ProductID      1353 non-null   int64  
 5   ProductName    1353 non-null   object 
 6   VendorID       1353 non-null   int64  
 7   ProductVendor  1353 non-null   object 
 8   UnitPrice      1353 non-null   float64
 9   Quantity       1353 non-null   int64  
 10  Subtotal       1353 non-null   float64
 11  Tax(8%)        1353 non-null   float64
 12  OrderTotal     1353 non-null   float64
dtypes: float64(4), int64(5), object(4)
memory usage: 137.5+ KB


In [6]:
# Change dtype for Date to datetime to perform calculations
df['Date'] = pd.to_datetime(df['Date'])
df.dtypes

TransactionID             int64
CustID                    int64
Date             datetime64[ns]
Employee                 object
ProductID                 int64
ProductName              object
VendorID                  int64
ProductVendor            object
UnitPrice               float64
Quantity                  int64
Subtotal                float64
Tax(8%)                 float64
OrderTotal              float64
dtype: object

### Grouping by Transactions

In [9]:
df_trans_amt = df.drop(['Date', 'ProductID', 'ProductName', 'VendorID', 'ProductVendor', 'CustID','Date'], axis=1)
df_trans_amt

TransactionID        Employee  UnitPrice  Quantity  Subtotal  Tax(8%)  \
0             10258   Nancy Davolio       25.6         6     153.6    12.29   
1             10275   Nancy Davolio        3.6        12      43.2     3.46   
2             10275   Nancy Davolio       44.0         6     264.0    21.12   
3             10292   Nancy Davolio       64.8        20    1296.0   103.68   
4             10293   Nancy Davolio       50.0        12     600.0    48.00   
...             ...             ...        ...       ...       ...      ...   
1348          11016  Anne Dodsworth       12.5        15     187.5    15.00   
1349          11016  Anne Dodsworth       19.0        16     304.0    24.32   
1350          11058  Anne Dodsworth       10.0         3      30.0     2.40   
1351          11058  Anne Dodsworth       34.0        21     714.0    57.12   
1352          11058  Anne Dodsworth       28.5         4     114.0     9.12   

      OrderTotal  
0         165.89  
1          46.66  
2         285.12  
3        1399.68  
4         648.00  
...          ...  
1348      202.50  
1349      328.32  
1350       32.40  
1351      771.12  
1352      123.12  

[1353 rows x 7 columns]

In [27]:
df_mean = df_trans_amt.groupby(['Employee']).mean().round(1)
df_mean

TransactionID  UnitPrice  Quantity  Subtotal  Tax(8%)  \
Employee                                                                  
Andrew Fuller           10717.8       27.9      12.6     342.0     27.4   
Anne Dodsworth          10736.4       32.9      13.3     406.7     32.5   
Janet Leverling         10647.7       24.7      13.0     292.6     23.4   
Laura Callahan          10631.8       25.0      13.1     331.5     26.5   
Margaret Peacock        10625.9       28.0      12.5     356.7     28.5   
Michael Suyama          10656.5       23.2      13.0     295.8     23.7   
Nancy Davolio           10699.9       26.1      11.3     314.6     25.2   
Robert King             10648.5       23.5      12.9     293.0     23.4   
Steven Buchanan         10626.2       21.4      13.8     292.8     23.4   

                  OrderTotal  
Employee                      
Andrew Fuller          369.3  
Anne Dodsworth         439.2  
Janet Leverling        316.0  
Laura Callahan         358.0  
Margaret Peacock       385.3  
Michael Suyama         319.4  
Nancy Davolio          339.8  
Robert King            316.5  
Steven Buchanan        316.3

In [28]:
df_sum = df_trans_amt.groupby(['Employee']).sum()
df_sum

TransactionID  UnitPrice  Quantity  Subtotal  Tax(8%)  \
Employee                                                                  
Andrew Fuller           1554078    4042.43      1829  49583.21  3966.66   
Anne Dodsworth           665659    2038.56       822  25215.25  2017.22   
Janet Leverling         2086955    4833.81      2552  57347.97  4587.85   
Laura Callahan          1828669    4291.63      2250  57014.73  4561.14   
Margaret Peacock        2815855    7411.25      3308  94531.50  7562.47   
Michael Suyama          1257471    2737.12      1532  34899.51  2791.96   
Nancy Davolio           2364681    5765.38      2495  69532.81  5562.63   
Robert King             1075497    2374.41      1299  29596.60  2367.73   
Steven Buchanan          775711    1560.15      1004  21377.75  1710.24   

                  OrderTotal  
Employee                      
Andrew Fuller       53549.87  
Anne Dodsworth      27232.47  
Janet Leverling     61935.82  
Laura Callahan      61575.87  
Margaret Peacock   102093.97  
Michael Suyama      37691.47  
Nancy Davolio       75095.44  
Robert King         31964.33  
Steven Buchanan     23087.99

In [29]:
df_sum_sorted = df_median.sort_values(by='Subtotal', ascending=False)
df_sum_sorted

TransactionID  UnitPrice  Quantity  Subtotal  Tax(8%)  \
Employee                                                                  
Margaret Peacock        2815855    7411.25      3308  94531.50  7562.47   
Nancy Davolio           2364681    5765.38      2495  69532.81  5562.63   
Janet Leverling         2086955    4833.81      2552  57347.97  4587.85   
Laura Callahan          1828669    4291.63      2250  57014.73  4561.14   
Andrew Fuller           1554078    4042.43      1829  49583.21  3966.66   
Michael Suyama          1257471    2737.12      1532  34899.51  2791.96   
Robert King             1075497    2374.41      1299  29596.60  2367.73   
Anne Dodsworth           665659    2038.56       822  25215.25  2017.22   
Steven Buchanan          775711    1560.15      1004  21377.75  1710.24   

                  OrderTotal  
Employee                      
Margaret Peacock   102093.97  
Nancy Davolio       75095.44  
Janet Leverling     61935.82  
Laura Callahan      61575.87  
Andrew Fuller       53549.87  
Michael Suyama      37691.47  
Robert King         31964.33  
Anne Dodsworth      27232.47  
Steven Buchanan     23087.99

In [10]:
total_monthly_sales = df_trans_amt.groupby(['Month'])['Subtotal'].sum().reset_index(name='Monthly Sales')

KeyError: 'Month'

In [ ]:
total_monthly_sales = df.groupby(['Month'])['Subtotal'].sum().reset_index(name='Monthly Sales')

total_monthly_sales_sorted = monthly_sales.sort_values(by='Monthly Sales', ascending=False).reset_index(drop=True)

total_monthly_sales

In [ ]:
quarterly_sales = df_trans_amt.groupby(['Quarter'])['Subtotal'].sum().reset_index(name='Quarterly Sales')

total_quarterly_sales_sorted = quarterly_sales.sort_values(by='Quarterly Sales', ascending=False).reset_index(drop=True)

total_quarterly_sales_sorted

In [ ]:
yearly_sales = df_trans_amt.groupby(['Year'])['Subtotal'].sum().reset_index(name='Yearly Sales')

total_yearly_sales_sorted = yearly_sales.sort_values(by='Yearly Sales', ascending=False).reset_index(drop=True)

total_yearly_sales_sorted